### ISSUE TITLE
[css-color] Adding names to the color previews

### ISSUE THREAD

**Issue by matatk** [w3c/csswg-drafts#7686]
*This comment is from the APA WG and relates to [our review of CSS Color Module Level 5](https://github.com/w3c/a11y-request/issues/29#issuecomment-1236389445).*

Adding "alternative text" to the visual color preview feature could be very helpful for people with vision impairments who currently have to rely on the colors' coordinates alone. Whilst coordinates are provided for every color mentioned in the examples, they may not give the reader much of a sense of what the color might look like, making the examples hard to follow (e.g. it's much easier to understand the effects of certain types of mixing by glancing at the previews and noting the similarities or differences from the input colors, rather than parsing through the coordinates). The document describes some of the colors that have previews, but not all.

There are some tools that gives you the closest color name to a given (usually hex-encoded) color, from an extensive palette of predefined colors. This sort of tool would give a good enough idea for most people who may be unable to perceive the colors directly. The color names could be generated as part of the document build process, and placed in visually-hidden text in the document. The text should also indicate if the colours are transparent.

One such tool is "color-to-name" (<https://github.com/stanleyfok/color-to-name>)—however, a small number of the color names in the palette may be considered vulgar, so they would either have to be filtered out, or a different tool selected. The same applies for color names that may be triggering for some people. We can do more research on tools of this nature, if you like.

It's noted that out-of-gamut colors are indicated by a red border on their previews, and they're always described in accompanying text as being out of gamut, which is helpful.

**Comment by svgeesus** [comment]
This might be interesting to try. I can see a few problems:

- offensive names.
- unclear names. 
- trademarked names
- commercial factors

There are some who would find [these](https://color-register.org/color/?q=vomit) sort of [names](https://color-register.org/color/?q=poop) offensive.

[gainsboro](https://color-register.org/color/?q=gainsboro) is just mysterious, [dodger blue](https://color-register.org/color/?q=dodger) is culturally specific.

[IBM Blue](https://color-register.org/color/?q=ibm) is a [branded color](https://usbrandcolors.com/ibm-colors/); same with [Netflix Red](https://color-register.org/color/?q=netflix) also [branded](https://usbrandcolors.com/netflix-colors/). All Pantone colors are covered by license agreements. Getting tangled in trademark law is not desirable.

the [Official Register of Color Names]() charges [USD 37](https://store.color-register.org/) per name.

**Comment by matatk** [comment]
Agreed regarding the problems, at least with this library—it doesn't seem like this is one we can really use. As also mentioned above, some may be triggering for some users.

I did find some other similar tools, but they didn't have as wide a gamut of colours. Maybe they would be sufficient though. There's one in particular that I found that worked in a similar way (with a narrower, but potentially less challenging, palette). I didn't mention it above, but should be able to dig it out of my history.

**Comment by svgeesus** [comment]
So, where are we on this? Auto-naming of colors looks like a little bit of a research project.

**Comment by matatk** [comment]
Agreed; if you're interested in pursuing the idea, if it's possible to find a suitable set of names, then I will have a look around for such as et of names as and when I'm able (this may take a while), and we can come back to this.

**Comment by svgeesus** [comment]
OK, closing for now until this option becomes more tractable.
